packages/datadog-plugin-aws-sdk/src/services/bedrockruntime/utils.js

```diff
@@ -350,14 +380,27 @@ function extractTextAndResponseReason (response, provider, modelName) {
         if (modelName.includes('embed')) {
           return new Generation({ message: body.embedding })
         }
-        const results = body.results || []
-        if (results.length > 0) {
-          const result = results[0]
+        if (body.results !== undefined) {
+          const results = body.results || []
+          if (results.length > 0) {
+            const result = results[0]
+            return new Generation({
+              message: result.outputText,
+              finishReason: result.completionReason,
+              inputTokens: body.inputTextTokenCount,
+              outputTokens: result.tokenCount
+            })
+          }
+        } else if (body.output !== undefined) {
+          const output = body.output || {}
           return new Generation({
-            message: result.outputText,
-            finishReason: result.completionReason,
-            inputTokens: body.inputTextTokenCount,
-            outputTokens: result.tokenCount
+            message: output.message?.content[0]?.text,
```

Comment:
```suggestion
            message: output.message?.content[0]?.text ?? 'Unsupported content type',
```

might be good to have a default if it doesn't have a text block but some other kind of block, but lmk what you think!

Reply:
Definitely agree, just updated this!